src/user/user.service.ts

```diff
@@ -224,4 +234,34 @@ export class UserService {
             }
         }
     }
+
+    public async requestOtp(
+        body: RequestOtpRequest
+    ): Promise<RequestOtpResponse> {
+        const { mobileNumber } = body;
+        const otp = Math.floor(Math.random() * 900000) + 100000;
+        const expiryTimestamp =
+            new Date().getTime() +
+            parseInt(this.config.get(ConfigSettings.OTP_EXPIRY_TIME, "60000"));
+        const salt = this.config.get(
+            ConfigSettings.OTP_HASHING_SALT,
+            "Hi i'm default salt from idem proxy :)"
+        );
+        const messageForHash = mobileNumber + otp + expiryTimestamp + salt;
+        const hash = crypto
+            .createHmac(
+                "sha256",
+                this.config.get(ConfigSettings.OTP_HASHING_SECRET)
+            )
+            .update(messageForHash)
+            .digest("hex");
+
+        const smsMessage = "Your OTP code for IDEM is " + otp;
+        await this.smsService.send(mobileNumber, smsMessage);
```

Comment:
I think we can just call it a message and can we use a template?

```suggestion
        const message = `Your OTP code for IDEM is ${otp}`;
        await this.smsService.send(mobileNumber, message);
```